cmd/client_test.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove (@evanSpendlove)
+//
+// Client_test tests the client.go file to ensure that the gRPC client for connecting
+// to Cloudprober works as expected. This client is used for adding and removing probes.
+
+package cmd
+
+import (
+	"github.com/golang/glog"
+	"os"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/cloudprober/examples/extensions/myprober/myprobe"
+	probes_configpb "github.com/google/cloudprober/probes/proto"
+	targetspb "github.com/google/cloudprober/targets/proto"
+	"github.com/google/cloudprober"
+	"github.com/google/cloudprober/web"
+
+	"github.com/googleinterns/step224-2020/hermes"
+)
+
+const (
+	cpCfg     string = "grpc_port:9314" // Cloudprober config
+	rpcServer string = "localhost:9314" // Cloudprober gRPC server address
+
+	testProbe0 string = "testExtension0"
+	testProbe1 string = "testExtension1"
+)
+
+//setupHermes initialises an instance of Cloudprober so that RPCs can be sent to it.
+func setupHermes(hermes *hermes.Hermes) {
+	err := hermes.InitialiseCloudproberFromConfig(cpCfg)
+	if err != nil {
+		glog.Fatalf("Cloudprober could not be initialised, err: %v", err)
+	}
+}
+
+// TestMain is used to setup Cloudprober before tests are run.
+// This is necessary for there to be a gRPC server to make RPCs to.
+func TestMain(m *testing.M) {
+	test_hermes := &hermes.Hermes{}
+
+	setupHermes(test_hermes)              // Set up Hermes and initialise Cloudprober
```

Comment:
It appears that (at least for now) setupHermes doesn't actually setup hermes but instead only initializes cloud prober.
```suggestion
	setupHermes(test_hermes)              // Set up Cloudprober
```

Reply:
I have refactored this method, so it is now `setupCloudproberAndClient`.